### PR TITLE
fix: fixed notation/test/e2e/suite/plugin

### PR DESCRIPTION
**Pull request by patrickzheng200**
A complie error occurred when dependabot bumping up notation/test/e2e/plugin: https://github.com/notaryproject/notation/pull/565.

Found the cause with @JeyJeyGao. It's due to notation/test/e2e/suite/plugin importing notation/test/e2e/plugin, which is not intended. This PR contains the quick fix for the issue.

### CODE REVIEW

**Review by JeyJeyGao — COMMENTED** [review]
LGTM

**Review by JeyJeyGao — DISMISSED** [review]
LGTM

File: test/e2e/go.mod [hunk 1]
```diff
@@ -3,19 +3,18 @@ module github.com/notaryproject/notation/test/e2e
...
 
 require (
-	github.com/notaryproject/notation-core-go v1.0.0-rc.1
-	github.com/notaryproject/notation/test/e2e/plugin v1.0.0
+	github.com/notaryproject/notation-core-go v1.0.0-rc.2
 	github.com/onsi/ginkgo/v2 v2.3.0
 	github.com/onsi/gomega v1.22.1
 	github.com/opencontainers/image-spec v1.1.0-rc2
-	oras.land/oras-go/v2 v2.0.0-rc.6
+	oras.land/oras-go/v2 v2.0.0
```
Comment (shizhMSFT) [inline test/e2e/go.mod:10]: The latest version is `v2.0.2`

**Review by JeyJeyGao — APPROVED** [review]
LGTM

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Review by priteshbandi — APPROVED** [review]
LGTM

**Pull request merged**